docs/static/monitoring-apis.asciidoc

```diff
@@ -0,0 +1,312 @@
+[[monitoring]]
+== Monitoring APIs
+
+Logstash provides the following monitoring APIs to retrieve runtime metrics
+about Logstash:
+
+* <<hot-threads-api>>
```

Comment:
@acchen97 Right now the APIs are listed in alphabetical order. One of my pet peeves about our documentation is that we don't always have a clear organizational pattern, which leaves it up to users to figure out where we've buried stuff. For reference info, I prefer to alphabetize the topics to make it easier for people to find stuff by name (which admittedly isn't important with a small set of topics). If we are planning to add a bunch of other monitoring APIs later, then I would prefer to keep the current organization and be consistent in all the APIs that we document. I'm open to your suggestion, though, if you feel strongly.